commands/http/handler.go

```diff
@@ -121,6 +121,13 @@ func (i internalHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	i.ctx.Context = ctx
 	req.SetContext(i.ctx)
 
+	if cn, ok := w.(http.CloseNotifier); ok {
+		go func() {
+			<-cn.CloseNotify()
```

Comment:
may want: 
```go
select {
case <-cn.CloseNotify():
  cancel()
case <-ctx.Done():
}
```
in case the CloseNotify never comes, or ctx is cancelled earlier (parent timeout or whatever)